lib/rate-limiter.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const _ = require('lodash');
+
+let rateTable = {};
+
+exports.register = (defaultRate, routes) => {
+  routes = routes.filter((route) => {
+    return (route.method === 'get' || route.method === 'post' || route.method === 'delete') && _.get(route, 'settings.plugins.rateLimit.enabled') !== false;
+  });
+
+  routes.forEach((route) => {
+    const rate = defaultRate;
+    const plugins = route.settings.plugins;
+
+    if (plugins && plugins.rateLimit) {
+      rate.limit = plugins.rateLimit.limit;
+      rate.window = plugins.rateLimit.window;
+    }
+
+    rateTable[`${route.method}:${route.path}`] = rate;
+  });
+
+  return rateTable;
+};
+
+exports.reset = () => {
```

Comment:
If you don't have `rateTable` set at the top, this function isn't necessary anymore.